src/lambdaconnect_sync/db_drivers/memory.cljc

```diff
@@ -0,0 +1,729 @@
+              true (assoc-in [:collections
+                              (:destination-entity relationship)
+                              :collection-content
+                              new-value
+                              (t/datomic-inverse-name inverse-relationship)]
+                             #{(id->rel entity-id)}))))))))
+
+(defmethod process-transaction-entry :db/retract
+  [[_ entity-id attribute-key retracted-value] 
+   snapshot 
+   datomic-relationships 
+   entities-by-id
+   allocated-ids]
+
+  (let [entity-name (entities-by-id entity-id)]
+    (assert entity-name)
+    (if-not (datomic-relationships attribute-key)
+      ;; Attribute
+      (update-in snapshot [:collections 
+                          entity-name 
+                          :collection-content 
+                          entity-id]
+                 #(do (assert (= retracted-value (attribute-key %)))
+                      (dissoc % attribute-key)))
+      ;; Relationship
+      (let [relationship (get-in snapshot [:entities-by-name
+                                           entity-name
+                                           :datomic-relationships
+                                           (name attribute-key)])
+            inverse-relationship (get-in snapshot [:entities-by-name
+                                                   (:inverse-entity relationship)
+                                                   :relationships
+                                                   (:inverse-name relationship)])
+            _ (assert relationship)
+            _ (assert inverse-relationship)
+
+            id->rel (fn [id]
+                      (let [uuid (get-in snapshot 
+                                         [:collections
+                                          (entities-by-id id)
+                                          :collection-content
+                                          id
+                                          :app/uuid])]
+                        (assert uuid)
+                        {:db/id id
+                         :app/uuid uuid}))
+            retracted-value (lookup-relationship snapshot allocated-ids relationship retracted-value)]
+        (cond 
+
+          ;; many to many 
+
+          (and (:to-many relationship)
+               (:to-many inverse-relationship))         
+          
+          (-> snapshot
+              (update-in [:collections
+                          entity-name
+                          :collection-content
+                          entity-id
+                          attribute-key]
+                         #(disj (or % #{}) (id->rel retracted-value)))
+              (update-in [:collections
+                          (:destination-entity relationship)
+                          :collection-content
+                          retracted-value
+                          (t/datomic-inverse-name inverse-relationship)]
+                         #(disj (or % #{}) (id->rel entity-id))))
```

Comment:
Jeśli `retracted-value` jest nilem, to odpinam wszystkie obiekty, z którym jestem połączony relacją.